akka-stream/src/main/scala/akka/stream/Graph.scala

```diff
@@ -69,6 +70,35 @@ trait Graph[+S <: Shape, +M] {
   def addAttributes(attr: Attributes): Graph[S, M] = withAttributes(traversalBuilder.attributes and attr)
 }
 
+object Graph {
+
+  /**
+   * Java API
+   * Transform the materialized value of this Flow, leaving all other properties as they were.
+   *
+   * @param g the graph being transformed
+   * @param f function to map the graph's materialized value
+   * @return a graph with same semantics as the given graph, except from the materialized value which is mapped using f.
+   */
+  def mapMaterializedValue[S <: Shape, M1, M2](g: Graph[S, M1])(f: M1 => M2): Graph[S, M2] = {
+    new GenericGraph(g.shape, g.traversalBuilder).mapMaterializedValue(f)
+  }
+
+  /**
+   * Scala API, see https://github.com/akka/akka/issues/28501 for discussion why this can't be an instance method on class Graph.
+   * @param self the graph whose material value will be mapped
```

Comment:
```suggestion
   * @param self the graph whose materialized value will be mapped
```

Reply:
Not sure we need scaladoc here, seems unlikely you would end up here? But doesn't hurt as well.